src/main/java/by/exadel/internship/service/InternshipService.java

```diff
@@ -37,16 +40,64 @@ public GuestInternshipDTO getById(UUID uuid) {
     }
 
     public List<GuestInternshipDTO> getAll() {
-
+        putClassNameInMDC();
         MDC.put("className", InternshipService.class.getSimpleName());
         log.info("Try to get all Internships");
-
         List<Internship> internships = internshipRepository.findAll();
         List<GuestInternshipDTO> guestInternshipDTOList = internShipMapper.map(internships);
-
+        log.info("Successfully got list of Internships");
         log.info("Successfully list of Internships");
-
         return guestInternshipDTOList;
 
     }
+
+    public List<GuestInternshipDTO> getAllDeleted() {
+        putClassNameInMDC();
+        log.info("Try to get deleted Internships");
+        List<Internship> deletedInternships = internshipRepository.findAllDeleted();
+        List<GuestInternshipDTO> guestDeletedInternshipDTOList = internShipMapper.map(deletedInternships);
+        log.info("Successfully got list of deleted Internships");
+        return guestDeletedInternshipDTOList;
+    }
+
+    public GuestInternshipDTO getDeletedInternshipById(UUID uuid) {
+        putClassNameInMDC();
+        log.info("Try to get deleted Internships with uuid= " + uuid);
+        Internship deletedInternship = internshipRepository.findDeletedById(uuid);
```

Comment:
```suggestion
        Optional<Internship> deletedInternship = internshipRepository.findDeletedById(uuid);
```

You can return Optional as the result, so null check will be more beautiful.

Reply:
Done